src/components/molecules/AuthenticationCard.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import AccentedLink from '../atoms/AccentedLink';
+import CardContainer from '../atoms/CardContainer';
+import Input from '../atoms/Input';
+import OutlinedButton from '../atoms/OutlinedButton';
+
+export enum AuthType {
+  Login = 'Sign in',
+  Register = 'Sign up',
+}
+
+export interface IOnLogin {
+  email: string;
+  password: string;
+}
+
+export interface IOnRegister extends IOnLogin {
+  firstName: string;
+  lastName: string;
+}
+
+interface IAuthenticationCard {
+  children?: never;
+  onRegister?: (args: IOnRegister) => void;
+  onLogin?: (args: IOnLogin) => void;
+  type: AuthType;
+  error?: string;
+}
+
+const AuthCardContainer = styled(CardContainer)`
+  margin: auto;
+  width: 30vw;
+  margin-left: 10vw;
+
+  &, form * {
+    background: ${props => props.theme.main};
+  }
+
+  h1 {
+    margin-bottom: .7em;
+  }
+
+  form>* {
+    margin-bottom: .6em;
+  }
+
+  form {
+    margin-bottom: .6em;
+  }
+
+  form~a {
+    display: block;
+  }
+`;
+
+const AuthenticationCard: React.FC<IAuthenticationCard> = props => {
+  const [email, setEmail] = useState('');
+  const [firstName, setFirstName] = useState('');
+  const [lastName, setLastName] = useState('');
+  const [password, setPassword] = useState('');
+
+  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    if (props.type === AuthType.Register) {
+      props.onRegister!({ email, password, firstName, lastName });
+    } else {
+      props.onLogin!({ email, password });
+    }
+  };
+
+  const nameInputs = props.type === AuthType.Register ? (
+    <>
+      <Input placeholder="first name" id="firstName" type="text" value={firstName} setState={setFirstName} />
+      <Input placeholder="surname" id="lastName" type="text" value={lastName} setState={setLastName} />
+    </>
+  ) : null;
+
+  return (
+    <AuthCardContainer>
+      <h1>{props.type}</h1>
+
+      {props.error && <p>{props.error}</p>}
+
+      <form onSubmit={handleSubmit}>
+        {nameInputs}
+        <Input placeholder="email" id="email" type="text" setState={setEmail} value={email} />
+        <Input placeholder="password" id="password" type="password" setState={setPassword} value={password} />
+        <OutlinedButton>{props.type}</OutlinedButton>
+        {props.type === AuthType.Login && <AccentedLink to="#">Forgot your password?</AccentedLink>}
+      </form>
+      <AccentedLink to={props.type === AuthType.Login ? '/register' : '/'}>
+        {props.type === AuthType.Login ? 'Don\'t have an account?' : 'Already have an account?'}
+      </AccentedLink>
+    </AuthCardContainer>
+  );
+};
+
+export default AuthenticationCard;
```

Comment:
Would you be so kind as to explain your reasoning behind demoting this component from an _organism_ to a _molecule_? My guessing is because you made it more generic, but I'd love to hear your reasoning. 😄 

In my opinion, this would be an organism because it makes up a very distinct part of the interface. I would think of lines 81-83 (the header + error message), 85-91 (the form) and 92-95 (the link), as a molecule, a molecule, and an atom, respectively, making the entire component an organism. 

These things are abstract concepts, but I believe the more we agree on the [Atomic Design](https://medium.com/@janelle.wg/atomic-design-pattern-how-to-structure-your-react-application-2bb4d9ca5f97) hierarchy, the easier it will be to collaborate in the time to come. 👥 👍

Reply:
Good point. The reason I moved it was mostly because of the name - `AuthenticationCard`. A card component shouldn't really carry that much logic, as it's - _often_ - purely presentational. I do in retrospect however, believe that this should be an _organism_, but should be renamed to something that conveys its purpose a bit clearer, as it's no longer simply a card container, but a fully fledged authenticatin page. E.g. `AuthenticationModule` or simply `Authentication`.